### PR TITLE
doc: add software maturity table

### PR DESCRIPTION
**Pull request by wiba-nordic**
Adds a table detailing the difference between supported
and experimental features.

Signed-off-by: Wille Backman <wille.backman@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/software_maturity.rst [hunk 1]
```diff
@@ -11,11 +11,54 @@ Following are the different software maturity levels shown for the different fea
...
+.. _software_maturity_table:
+
+.. list-table:: Software maturity
+   :header-rows: 1
+   :align: center
+
+   * - 
+     - Supported
+     - Experimental
+     - Unsupported
```
Comment (peknis) [inline doc/nrf/software_maturity.rst:27]: Should this be "Not supported" as you have that just above the table?

File: doc/nrf/software_maturity.rst [hunk 2]
```diff
@@ -11,11 +11,54 @@ Following are the different software maturity levels shown for the different fea
...
+   * - Technical support
+     - Customer issues raised for features supported for development on tagged |NCS| releases are handled by technical support on DevZone.
+     - Customer issues raised for experimental features on tagged |NCS| releases are handled by technical support on DevZone.
+     - Not available.
+   * - Bug fixing
+     - Reported critical bugs will be resolved in both main and latest tagged version.
+     - Bug fixes and improvements are not guaranteed to be applied to current versions.
+     - Not available .
```
Comment (peknis) [inline doc/nrf/software_maturity.rst]:
```suggestion
     - Not available.
```

File: doc/nrf/software_maturity.rst [hunk 3]
```diff
@@ -11,11 +11,54 @@ Following are the different software maturity levels shown for the different fea
...
+   * - Implementation completeness
+     - Complete implementation of the agreed features set.
+     - Significant changes may be applied version to version.
+     - A feature or component is "supported" as an implementation, but is not compatible with, or tested on a specific device or series of products.
```
Comment (peknis) [inline doc/nrf/software_maturity.rst:39]: Dunno whether the quotation marks are needed.

File: doc/nrf/software_maturity.rst [hunk 4]
```diff
@@ -11,11 +11,54 @@ Following are the different software maturity levels shown for the different fea
...
+   * - API definition
+     - API definition is stable
```
Comment (peknis) [inline doc/nrf/software_maturity.rst]:
```suggestion
     - API definition is stable.
```

File: doc/nrf/software_maturity.rst [hunk 5]
```diff
@@ -11,11 +11,54 @@ Following are the different software maturity levels shown for the different fea
...
+
+       A feature or component that is in in a core supported sub-system but is in development and either:
+
+         * Not fully tested according to existing test plans.
+         * In development and changing; not complete.
+     - Not applicable.
 
+   * - Verification
+     - Verified for product development.
+     - Incomplete verification
```
Comment (peknis) [inline doc/nrf/software_maturity.rst:60]:
```suggestion
     - Incomplete verification.
```

File: doc/nrf/software_maturity.rst [hunk 6]
```diff
@@ -11,11 +11,54 @@ Following are the different software maturity levels shown for the different fea
...
+   * - API definition
+     - API definition is stable
+     - API definition may change going to production.
+     - Not available.
+   * - Maturity
+     - Suitable for integration in end-product development.
+
+       A feature or component that is either fully complete on first commit, or has previously been labelled "experimental" is ready for use in end-product projects.
+
+     - Suitable for prototyping an evaluation.
```
Comment (peknis) [inline doc/nrf/software_maturity.rst:49]:
```suggestion
     - Suitable for prototyping or evaluation.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/software_maturity.rst [hunk 7]
```diff
@@ -11,11 +11,54 @@ Following are the different software maturity levels shown for the different fea
...
+       A feature or component that is in in a core supported sub-system but is in development and either:
+
+         * Not fully tested according to existing test plans.
+         * In development and changing; not complete.
+     - Not applicable.
 
+   * - Verification
+     - Verified for product development.
+     - Incomplete verification
+     - Not available.
```
Comment (peknis) [inline doc/nrf/software_maturity.rst:61]:
```suggestion
     - Not applicable.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Comment by wiba-nordic** [comment]
Added a little bit of formatting to the table.

**Review by peknis — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**